background.ts

```diff
@@ -216,10 +217,16 @@ function installDevtools() {
 }
 
 async function doFirstRun() {
-  if (!settings.isFirstRun()) {
+  if (!settings.firstRunNeeded()) {
+    waitForFirstRunCompletion = false;
+
     return;
   }
-  await window.openFirstRun();
+  try {
+    await window.openFirstRun();
+  } finally {
+    waitForFirstRunCompletion = false;
+  }
 }
```

Comment:
Simpler version:

```
async function doFirstRun() {
  if (settings.firstRunNeeded()) {
    try {
      await window.openFirstRun();
    }
  }
  waitForFirstRunCompletion = false;
}

Reply:
FWIW, I would `s/FirstRun/FirstRunDialog`:

```ts
async function doFirstRunDialog() {
  if (settings.firstRunDialogNeeded()) {
    try {
      await window.openFirstRunDialog();
    }
  }
  waitForFirstRunDialogCompletion = false;
}
```

Because to me the "first run" refers to the app as a while, and not just the initial dialog.